tests/api_test.py

```diff
@@ -625,6 +625,24 @@ def test_device_put_across_platforms(self):
       x = api.device_put(val, device=cpu_device)
       self.assertEqual(x.device_buffer.device(), cpu_device)
 
+  def test_device_put_sharded(self):
+    devices = api.local_devices()
+    n_devices = len(devices)
+    x = [np.arange(i, i + 4) for i in range(n_devices)]
+    y = api.device_put_sharded(x, devices)
+    self.assertEqual(len(y.device_buffers), len(devices))
+    self.assertTrue(all(b.device() == d for b, d in zip(y.device_buffers, devices)))
+    self.assertAllClose(y, jnp.stack(x))
+
+  def test_device_put_replicated(self):
+    devices = api.local_devices()
+    n_devices = len(devices)
+    x = np.arange(4)
+    y = api.device_put_replicated(x, devices)
+    self.assertEqual(len(y.device_buffers), len(devices))
+    self.assertTrue(all(b.device() == d for b, d in zip(y.device_buffers, devices)))
+    self.assertAllClose(y, jnp.stack(n_devices * [x]))
+
```

Comment:
Can you also add a pytree test for put_sharded and put_replicate? I think testing a single pytree for each should be sufficient.

Reply:
Done.